codegen/templates/http_client.tmpl

```diff
@@ -123,19 +124,32 @@ func {{$exportName}}(deps *module.Dependencies) Client {
 	if deps.Default.Config.ContainsKey("clients.{{$clientID}}.circuitBreakerDisabled") {
 		circuitBreakerDisabled = deps.Default.Config.MustGetBoolean("clients.{{$clientID}}.circuitBreakerDisabled")
 	}
+
+	//get mapping of client method and it's timeout
+	//if mapping is not provided, use client's timeout for all methods
+	clientMethodTimeoutMapping := make(map[string]int64)
+	if deps.Default.Config.ContainsKey("clients.{{$clientID}}.methodTimeoutMapping") {
+		deps.Default.Config.MustGetStruct("clients.{{$clientID}}.methodTimeoutMapping", &clientMethodTimeoutMapping)
+	} else {
+		for _, serviceMethodName := range methodNames {
+			methodName := strings.Split(serviceMethodName, "::")[1]
```

Comment:
IMO `strings.Split` shouldn't be in the hot path. It performs not so efficiently for a simple use case and this code needs to be executed against every RPC. We should rather prefer to write our own if you are already aware of the format.

Here's where you can see the allocation.
```
a := make([]string, n)
```

<img width="817" alt="image" src="https://user-images.githubusercontent.com/12872673/198103887-900573e2-3e01-4a57-b2a4-bd771c3d5447.png">

Additionally: If you'd rather use `strings.Split` ensure you pass `strings.Split(string, separator, n = 1) so that we discard the rest of the string as soon as we find the split.